sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
Actually, I guess you have to use a metaclass. See https://docs.python.org/3.5/reference/datamodel.html#customizing-instance-and-subclass-checks.